### ISSUE TITLE
`echo "$foo" | ...` should become `... <<< "$foo"` in many cases

### ISSUE THREAD

**Issue by Nightfirecat** [koalaman/shellcheck#956]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
foo='The quick
brown fox
jumps over
the lazy dog'
echo "$foo" | grep 'a'
```

#### Here's what shellcheck currently says:
(nothing)

#### Here's what I wanted or expected to see:
```
In bar.sh line 6:
echo "$foo" | grep 'a'
              ^-- SC20XX: Pass variable contents through stdin using <<<.
```

---

Of note, this will not apply to all builtins... Not all of them will process input through stdin. `grep` is used as an easy example here that I found sprinkled throughout some of my own scripts.

**Comment by Nightfirecat** [comment]
Similarly, `cat file | grep ...` or its variations (`head file | ...`, `tail file | ...`, etc.) would be better as `grep ... < file`.

```
$ echo 'The quick brown fox jumps over the lazy dog' > test.txt
$ time for ((i=0; i<100; i++)); do cat test.txt | grep -q 'a'; done

real    0m20.075s
user    0m5.022s
sys     0m9.404s

$ time for ((i=0; i<100; i++)); do grep -q 'a' < test.txt; done

real    0m12.529s
user    0m2.436s
sys     0m4.319s
```

**Comment by Pawamoy** [comment]
Well there is a warning for useless use of cat: https://github.com/koalaman/shellcheck/wiki/SC2002

**Comment by nihilus** [comment]
@Nightfirecat Rewriting code to work like that does make it take time to port it to another shell.

**Comment by Nightfirecat** [comment]
Perhaps related: (not sure if this should be a separate issue)

Some kind of issue should be raised on useless `echo`s like the following: (yes, this is a real example)

```
echo | openssl s_client -connect ...
```

**Comment by nihilus** [comment]
@Nightfirecat actually that is the same as 'echo -e "\\n"' and as such its makes sense. 

Again, avoid bashisms like '<<<', is just syntatic sugar.

**Comment by koalaman** [comment]
I've seen code where the `\n` matters, like

    echo | tee -a logfile
    var=$(echo | awk '{print 1.2+34}')

but also where it doesn't like (presumably) here.
